File: chazm-model/src/main/java/runtimemodels/chazm/model/notification/Publisher.java
package runtimemodels.chazm.model.notification;

/**
 * The {@linkplain Publisher} interface defines the APIs a publish/subscribe pattern (a.k.a. observer pattern). Publishers push out updates that get sent to
 * subscribers.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public interface Publisher {

    /**
     * Post an event.
     *
     * @param event the event.
     */
    <T> void post(final T event);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/notification/Publisher.kt
package runtimemodels.chazm.model.notification

/**
 * The [Publisher] interface defines the APIs a publish/subscribe pattern (a.k.a. observer pattern). Publishers push out updates that get sent to
 * subscribers.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
interface Publisher { // TODO to be replaced with RxJava

    /**
     * Post an event.
     *
     * @param <T>   the type of event.
     * @param event the event.
    </T> */
    fun <T : Any> post(event: T)

}
